src/suites/cts/validation/setVertexBuffers.spec.ts

```diff
@@ -0,0 +1,184 @@
+export const description = `
+setVertexBuffers validation tests.
+`;
+
+import { TestGroup } from '../../../framework/index.js';
+
+import { ValidationTest } from './validation_test.js';
+
+class F extends ValidationTest {
+  async init(): Promise<void> {
+    await Promise.all([super.init(), this.initGLSL()]);
+  }
+
+  getVertexBuffer(): GPUBuffer {
+    return this.device.createBuffer({
+      size: 256,
+      usage: GPUBufferUsage.VERTEX,
+    });
+  }
+
+  createRenderPipeline(bufferCount: number): GPURenderPipeline {
+    const descriptor: GPURenderPipelineDescriptor = {
+      vertexStage: this.getVertexStage(bufferCount),
+      fragmentStage: this.getFragmentStage(),
+      layout: this.getPipelineLayout(),
+      primitiveTopology: 'triangle-list',
+      colorStates: [{ format: 'rgba8unorm' }],
+      vertexInput: {
+        vertexBuffers: [
+          {
+            stride: 3 * 4,
+            attributeSet: [],
+          },
+        ],
+      },
+    };
+    for (let i = 0; i < bufferCount; i++) {
+      descriptor.vertexInput!.vertexBuffers![0].attributeSet.push({
+        format: 'float3',
+        shaderLocation: i,
+      });
+    }
+    return this.device.createRenderPipeline(descriptor);
+  }
+
+  getVertexStage(bufferCount: number): GPUProgrammableStageDescriptor {
+    const code = `
+      #version 450
+      ${[...new Array(bufferCount)]
```

Comment:
... and then can use that helper here too.

Reply:
Done